examples/client/Locomotion/src/context/newRideContext/index.tsx

```diff
@@ -219,13 +219,19 @@ const RidePageContextProvider = ({ children }: {
       const formattedEstimations = formatEstimations(services, estimations, tags);
       setChosenService(formattedEstimations.find((e: any) => e.eta));
       setServiceEstimations(formattedEstimations);
+      setIsLoading(false);
     } catch (e) {
+      setIsLoading(false);
       setServiceRequestFailed(true);
       setIsReadyForSubmit(false);
-    } finally {
-      setIsLoading(false);
     }
   };
+  useEffect(() => {
```

Comment:
shouldn't be related to the ride page context, please handle it in the component